File: course_101/exercises/easy_7/multiplicative_average.rb
# Write a method that takes an Array of integers as input, multiplies all the numbers together, divides the result by the number of entries in the Array,
# and then prints the result rounded to 3 decimal places. Assume the array is non-empty.

# solution 1
def show_multiplicative_average(arr)
  multiplicative_avarage = arr.reduce(:*) / arr.count.to_f
  
  format('And the result is %.3f', multiplicative_avarage)
end


puts show_multiplicative_average([3, 5])
# => The result is 7.500

puts show_multiplicative_average([6])
# => The result is 6.000

puts show_multiplicative_average([2, 5, 7, 11, 13, 17])
# => The result is 28361.667

File: course_js210_computational_thinking_and_javascript_programming/exercises/js210_small_problems/easy_4/multiplicative_average.js
// Write a function that takes an array of integers as input, multiplies all of the integers together, divides the result by the number of entries in the array, and returns the result as a string with the value rounded to three decimal places.

function showMultiplicativeAverage(numbers) {
  let multiplied = numbers.reduce((result, multiplicand) => result * multiplicand);
  let multipliedAverage = multiplied / numbers.length;

  return Number(multipliedAverage).toFixed(3);
}


console.log(showMultiplicativeAverage([3, 5]));                   // "7.500"
console.log(showMultiplicativeAverage([2, 5, 7, 11, 13, 17]));    // "28361.667"